examples/client/object_server.c

```diff
@@ -224,6 +224,50 @@ static uint8_t prv_get_value(lwm2m_data_t * dataP,
     }
 }
 
+static void remove_optional_resource(uint16_t *resList, int val, int id, int *nbRes) {
```

Comment:
It seems to me like `id` will always be positive and small:
```suggestion
static void remove_optional_resource(uint16_t *resList, int val, const uint_fast8_t id, int *nbRes) {
```

Reply:
Resource ID is 16 bit.